packages/devtools_app/lib/src/debugger/file_search.dart

```diff
@@ -114,22 +135,23 @@ List<ScriptRef> findMatches(
   List<ScriptRef> scriptRefs,
 ) {
   if (query.isEmpty) {
-    takeTopMatches(scriptRefs);
+    return scriptRefs;
   }
 
-  final exactMatches = scriptRefs
-      .where((scriptRef) => scriptRef.uri.caseInsensitiveContains(query))
-      .toList();
+  final exactMatches = [];
+  final fuzzyMatches = [];
 
-  if (exactMatches.length >= numOfMatchesToShow) {
-    return takeTopMatches(exactMatches);
+  for (final scriptRef in scriptRefs) {
+    final fullPath = scriptRef.uri;
+    final fileName = scriptRef.uri.split('/').last;
```

Comment:
to optimize the performance of this process, catch a list of split script URIs or fileNames rather than splitting every time.